### PR TITLE
Increase sid security

### PR DESCRIPTION
**Pull request by emveepee**
Add sid on Live TV.  Allow files instead of URLs for recordings.

### CODE REVIEW

**Comment by emveepee** [comment]
@joshuaboniface still not clear on the release and review process .  I submitted this in January and again zero feedback.  This security change is required for the next version of NextPVR and it would be nice to see it included before the release

File: Jellyfin.Plugin.NextPVR/Configuration/configPage.html [hunk 1]
```diff
@@ -103,6 +109,7 @@
                     page.querySelector('#chkDebugLogging').checked = config.EnableDebugLogging;
                     page.querySelector('#chkNewEpisodes').checked = config.NewEpisodes;
                     $('#selRecDefault', page).val(config.RecordingDefault);
+                    $('#selRecTransport', page).val(config.RecordingTransport);
```
Comment (oddstr13) [inline Jellyfin.Plugin.NextPVR/Configuration/configPage.html:112]: jQuery may or may not still be working, this needs testing, and most likely needs replacing by vanilla JS (`querySelector`).
I'd consider that out of scope for this PR however.

**Review by oddstr13 — APPROVED** [review]
Apologies, looks like I had this repo's notifications set to mentions only.

I have not tested these changes, but they look good on the surface, except from the fact that jQuery is still in general use in this plugin.

File: Jellyfin.Plugin.NextPVR/LiveTvService.cs [hunk 2]
```diff
@@ -495,7 +495,7 @@ public async Task<MediaSourceInfo> GetChannelStream(string channelOid, string me
             var baseUrl = Plugin.Instance.Configuration.WebServiceUrl;
             _liveStreams++;
 
-            string streamUrl = string.Format("{0}/live?channeloid={1}&client=jellyfin.{2}", baseUrl, channelOid, _liveStreams.ToString());
+            string streamUrl = string.Format("{0}/live?channeloid={1}&client=jellyfin.{2}&sid=", baseUrl, channelOid, _liveStreams.ToString(), Sid);
```
Comment (cvium) [inline Jellyfin.Plugin.NextPVR/LiveTvService.cs:498]: You're not actually setting the Sid?

Comment (oddstr13) [reply Jellyfin.Plugin.NextPVR/LiveTvService.cs:498]: Good catch, I didn't notice that.

**Comment by emveepee** [comment]
The Sid is poorly named,  but it is global to the addon and it is used for all API calls.  It is created on login https://github.com/jellyfin/jellyfin-plugin-nextpvr/blob/c593a946f6493fc3e8505f2aca389f9438446c14/Jellyfin.Plugin.NextPVR/LiveTvService.cs#L103 and there is a heartbeat call the keeps it active while the server is up.   https://github.com/jellyfin/jellyfin-plugin-nextpvr/blob/c593a946f6493fc3e8505f2aca389f9438446c14/Jellyfin.Plugin.NextPVR/LiveTvService.cs#L683

If the server is stopped the token is saved server side but there is additional reconnect logic to acquire a new one if it is invalid https://github.com/jellyfin/jellyfin-plugin-nextpvr/blob/c593a946f6493fc3e8505f2aca389f9438446c14/Jellyfin.Plugin.NextPVR/LiveTvService.cs#L79

**Comment by cvium** [comment]
No one asked or commented on the Sid name, intent or otherwise. You've added sid to the query parameters but forgot to add `{3}` after `sid=`

**Comment by emveepee** [comment]
> 
> 
> No one asked or commented on the Sid name, intent or otherwise. You've added sid to the query parameters but forgot to add `{3}` after `sid=`.

Right I see now that is a good catch,  a copy and paste issue.   Updated.

> 
> Which I suppose also begs the question: Did you test it properly?

Clearly no.  I only tested this in Emby and only recently where I actually do have it set correctly.  In January it was unit tested but it was only on spec at the time and the missing sid was not a requirement.

Unfortunately the changes to the addon from core in 10.7.0 make it impossible to diff Emby and Jellyfin

**Review by oddstr13 — APPROVED** [review]

**Review by h1dden-da3m0n — APPROVED** [review]

**Pull request merged**